### PR TITLE
Upgrade fp-ts library to 1.19.x

### PR DESCRIPTION
**Pull request by gunzip**

### CODE REVIEW

**Comment by pietro909** [comment]
@gunzip please note that [tests are failing](https://app.circleci.com/pipelines/github/pagopa/io-dev-api-server/1174/workflows/ab7d33d6-c451-4644-bb3d-d3d9416d7440/jobs/1184/steps?invite=true#step-107-8).
Ran locally the same test and works fine.
The error seems to happen [in utils/server.ts](https://github.com/pagopa/io-dev-api-server/pull/116/files#diff-dc76e9fe395862a856441b37a3fff3bf1708978f60ff2ea2d8281de63c93de16R36) at a line which wasn't affected by this PR 🤷‍♀️  
You may wanna investigate a bit?

**Comment by gunzip** [comment]
It looks like test are failing because the master is not aligned to CGN specs which is beyond the scope of this PR.
Once the master gets fixed I can backport the changes here.

**Comment by gunzip** [comment]
I'm digging a little bit more and it looks it has something to do with how circleci run tests and faker imports.

I've noticed this disclaimer on the new faker repo:

> This project was originally created and hosted at https://github.com/marak/Faker.js/ - however around 4th Jan, 2022 - the author decided to delete the repository (for unknown reasons).

I'm still checking..

**Comment by Undermaken** [comment]
I also did some changes on circleci config (invalidate dependencies cache) but it seems not working
@CrisTofani also noticed [his PR](https://github.com/pagopa/io-dev-api-server/pull/) runs tests without any errors

**Comment by pietro909** [comment]
@gunzip if you haven't heard about faker.js _yet_, I suggest you explore the rabbit's hole in your spare time for it is unfathomably deep. It shouldn't be that because we have a lockfile pinning it at 5.5.7

**Review by lucacavallaro — APPROVED** [review]
LGTM

nice work ✨

File: src/utils/server.ts [hunk 1]
```diff
@@ -1,22 +1,27 @@
...
 export const serverHostname = "0.0.0.0"; // public
 
 const interestingNetworkInterfaces = new Set(["en0", "eth0"]);
 
-const getIpv4Addresses = (nets: ReadonlyArray<os.NetworkInterfaceInfo>) =>
-  nets.filter(net => net.family === "IPv4" && !net.internal);
+const getIpv4Addresses = (
+  nets: ReadonlyArray<os.NetworkInterfaceInfo>
+): ReadonlyArray<os.NetworkInterfaceInfo> =>
+  nets.filter(net => net.family !== "IPv4" && !net.internal);
```
Comment (Undermaken) [inline src/utils/server.ts:14]:
```suggestion
  nets.filter(net => net.family === "IPv4" && !net.internal);
```
Schermata 2022-01-12 alle 18 04 47

**Review by Undermaken — CHANGES_REQUESTED** [review]

**Review by Undermaken — APPROVED** [review]
@gunzip thank you 🎸 
@pietro909 waiting your approval to merge

**Pull request merged**